.github/workflows/main.yml

```diff
@@ -18,24 +18,42 @@ jobs:
   validate:
     name: Validate 
     runs-on: ubuntu-latest
-    # Pull requests from the same repository won't trigger this checks as they were already triggered by the push
-    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
     steps:
-      - name: Clone the code
-        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
-        with:
+     - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+     - name: Cache protobuf library.
+       id: cache-protobuf
+       uses: actions/cache@d9747005de0f7240e5d35a68dca96b3f41b8b340
+       with:
+         path: protobuf
+         key: ${{ runner.os }}-protobuf
+     - name: Build protobuf library.
+       if: steps.cache-protobuf.outputs.cache-hit != 'true'
+       run: |
+         git clone https://github.com/protocolbuffers/protobuf.git
```

Comment:
ideally we'd pin this dependency. But I suppose we need to be in sync with BQ which we don't control.